e2e/_suites/fleet/features/fleet_mode_agent.feature

```diff
@@ -40,6 +40,17 @@ Examples:
 | centos |
 | debian |
 
+@upgrade-agent
+Scenario Outline: Upgrading the installed <os> agent
+  Given a "<os>" agent "stale" is deployed to Fleet with "tar" installer
+    And certs for "<os>" are installed
```

Comment:
is there value in leaving this separate?  you could maybe just copy the Go code into the end of the above installer step if we're never ever going to run tests for this feature without certs (I guess that would be a negative test to prove it fails? maybe not helpful...)

Reply:
i wanted to make it separate as having this in installer would mean that it would get executed every time installer is prepared and usually there's no need for that, having this separate dont slow down tests

Reply:
that is sound argument, but I don't see the 'Given' statement used elsewhere. :)  it is ok, no harm done, lets leave it as you've written

Reply:
I'd advocate for hiding this technical detail from the specification, and handle it in code, otherwise we will be polluting a reader's mind when understanding the behavior.

OTOH great job here @michalpristas !!!